src/controllers/loginController.ts

```diff
@@ -0,0 +1,79 @@
+import express , {Request, Response} from 'express';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import passport from 'passport';
+import { Strategy as LocalStrategy } from 'passport-local';
+import dotenv from 'dotenv';
+import { testPassword } from '../utils/validations';
+import Database from '../database/db';
+import sendResponse from '../utils/response';
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+dotenv.config()
+const router = express.Router();
+router.use(passport.initialize());
+router.use(express.json())
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+async function validateUser(email:string, password:string, done:Function) {
+    try {
+
+        //checking if user with this email exists in database
+
+        const user = await Database.User.findOne({
+            where: { email }
+        });
+
+        //comparing password entered with password stored in database
+
+        if (!user || !(await bcrypt.compare(password, user.password))) {
+            return done(null, false, { message: "User not found" });
+        }
+        return done(null, user);
+    } catch (err) {
+        return done(err);
+    }
+}
+
+
+
+passport.use(new LocalStrategy( { usernameField: 'email' },validateUser));
+
+
+
+const validatingUser = async(req:Request, res: Response) => {
+        passport.authenticate('local', { session: false }, async (err:any, user: any) => {
+        try {
+            
+            //testing the credentials entered in the request body
+
+            const passwordError = testPassword(req.body.password);
+            
+
+            if (passwordError) {
+              return sendResponse<null>(res, 400, null, passwordError);
+            }
+            
+            if (err) return (err);
+            if(!user) return sendResponse<null>(res, 400, null, 'User not found')
+            const tokenPayload = { email: user.email, password: user.password};
+            const token = jwt.sign(tokenPayload, process.env.ACCESS_TOKEN!);
+
+            res.cookie("token", token);
+            res.header('Authorization', `Bearer ${token}`);
+            return sendResponse<string>(res, 200, token, "Logged In Successfully");
+
+        } catch (err: any) {
+            res.status(401).json({error: err.issues[0].message})
+            return(err);
+        }
+    })(req, res, );
+}
```

Comment:
what is the difference between this variables

Reply:
passwordError: it validates  password from request body.
tokenPayload,token are dealing with generating token once credentials are correct